### PR TITLE
WIP: Upgrade lightstep exporter to OTel-Go 0.6 dependency

### PR DESCRIPTION
**Pull request by jmacd**
**Description:** 
This incorporates several fixes in the OTel-Go-to-LS exporter that is used in the collector exporter.
Re-do of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/249

**Link to tracking Issue:** n/a

**Testing:** pending

**Documentation:** n/a

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1237,6 +1272,14 @@ go.opentelemetry.io/collector v0.3.1-0.20200522130256-daf2fc71ac65 h1:ObXizS60v/
 go.opentelemetry.io/collector v0.3.1-0.20200522130256-daf2fc71ac65/go.mod h1:rcMDg+njfS22WiT/MioZMNnGyXrxfsKXA2d6xxpnnfc=
 go.opentelemetry.io/otel v0.2.3 h1:o97YpRYk0PyhCyuanlJY0DepUgAlyzl3rJ+4kb+456c=
 go.opentelemetry.io/otel v0.2.3/go.mod h1:OgNpQOjrlt33Ew6Ds0mGjmcTQg/rhUctsbkRdk/g1fw=
+<<<<<<< HEAD
```
Comment (tigrannajaryan) [inline go.sum:1275]: Merge conflict remnants, I guess :-)
I wonder why the build doesn't fail.

File: go.sum [hunk 2]
```diff
@@ -520,6 +522,13 @@ github.com/golang/protobuf v1.3.4 h1:87PNWwrRvUSnqS4dlcBU/ftvOIBep4sYuBLlh6rX2wk
 github.com/golang/protobuf v1.3.4/go.mod h1:vzj43D7+SQXF/4pzW/hwtAqwc6iTitCiVSaWz5lYuqw=
 github.com/golang/protobuf v1.3.5 h1:F768QJ1E9tib+q5Sc8MkdJi1RxLTbRcTf8LJV56aRls=
 github.com/golang/protobuf v1.3.5/go.mod h1:6O5/vntMXwX2lRkT1hjjk0nAC1IDOTvTlVgjlRvqsdk=
+github.com/golang/protobuf v1.4.0-rc.1/go.mod h1:ceaxUfeHdC40wWswd/P6IGgMaK3YpKi5j83Wpe3EHw8=
```
Comment (tigrannajaryan) [inline go.sum:525]: Can you please do a `go mod tidy`?

File: exporter/lightstepexporter/go.mod [hunk 3]
```diff
@@ -5,10 +5,10 @@ go 1.14
 require (
 	github.com/census-instrumentation/opencensus-proto v0.2.1
 	github.com/golang/protobuf v1.3.5
-	github.com/lightstep/opentelemetry-exporter-go v0.1.5
+	github.com/lightstep/opentelemetry-exporter-go v0.6.2
 	github.com/stretchr/testify v1.5.1
-	go.opentelemetry.io/collector v0.3.1-0.20200522130256-daf2fc71ac65
-	go.opentelemetry.io/otel v0.2.3
+	go.opentelemetry.io/collector v0.3.1-0.20200518164231-3729dac06f74
```
Comment (tigrannajaryan) [inline exporter/lightstepexporter/go.mod:10]: Why is this going to an older version?

File: go.sum (hunk 1 above)
Comment (jmacd) [reply go.sum]: I was confused by that. (Still am.)

File: go.sum (hunk 2 above)
Comment (jmacd) [reply go.sum]: Done.

File: exporter/lightstepexporter/go.mod (hunk 3 above)
Comment (jmacd) [reply exporter/lightstepexporter/go.mod]: When the build is broken, Go does very odd things in the .mod files, and then after you fix the build, you're left with these artifacts. Fixed.

**Comment by jmacd** [comment]
This is not passing a manual test at this point.

**Comment by tigrannajaryan** [comment]
Please fix codecov check and resolve conflicts.

**Comment by jmacd** [comment]
I can't tell what the codecov check is asking me for. Are we requiring an increase in coverage? There are no functional changes here.

**Comment by tigrannajaryan** [comment]
@bogdandrutu I think you setup the codecov here. Please help clarify what it complains about, I am not sure I understand.

**Comment by andrewhsu** [comment]
chatting with @jmacd and @codeboten, @codeboten will have a look into carrying this PR to help out

**Comment by jmacd** [comment]
We added tests just to silence the codecov report. We manually tested this. PTAL.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**